library/core/src/iter/range.rs

```diff
@@ -551,6 +565,18 @@ impl<A: Step> Iterator for ops::Range<A> {
     fn max(mut self) -> Option<A> {
         self.next_back()
     }
+
+    #[inline]
+    unsafe fn __iterator_get_unchecked(&mut self, idx: usize) -> Self::Item
+    where
+        Self: TrustedRandomAccess,
+    {
+        // Safety: The TrustedRandomAccess contract requires that callers only  pass an index
+        // that is in bounds.
+        // Additionally Self: TrustedRandomAccess is only implemented for Copy types
+        // which means even repeated reads of the same index would be safe.
```

Comment:
Is this actually required? `self.start` is cloned here anyway, so repeated calls should be fine even if the items are not `Copy`. Calling this should have no effects at all on `self`.

Reply:
A clone implementation could observe repeated clones of the same value rather than what the normal next implementation does.